homeassistant/components/discoverable.py

```diff
@@ -0,0 +1,121 @@
+"""Component to allow discovering Home Assistant on local network."""
+import json
+import logging
+import select
+import socket
+import threading
+from urllib.parse import urlsplit
+
+from homeassistant.const import (
+    __version__, EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+from homeassistant import remote
+
+DOMAIN = 'discoverable'
+DEPENDENCIES = ['api']
+
+MCAST_IP = "224.0.0.123"
+MCAST_PORT = 38123
```

Comment:
This is in the linux ephemeral range, which I think means we could have a clash. I'm not super sure about the impact of this on the multicast membership, as I can bind a tcp socket to 38123, which doesn't fail or interfere with the discovery.

AFAIK, UPNP uses a well-known (and assigned) port to do the initial discovery, and only does the random-port thing  for the actual interface, post-discovery.